dev/core/src/com/google/gwt/core/linker/SymbolMapsLinker.java

```diff
@@ -313,13 +326,46 @@ public ArtifactSet link(TreeLogger logger, LinkerContext context,
               }
             }
             // TODO(cromwellian): apply insert and remove edits
-            sourceMapGenerator.mergeMapSection(totalPrefixLines, 0, sourceMapString,
-                new ExtensionMergeAction() {
-                  @Override
-                  public Object merge(String extKey, Object oldVal, Object newVal) {
-                    return newVal;
-                  }
+            sourceMapGenerator.setStartingPosition(totalPrefixLines, 0);
+            SourceMapConsumerV3 section = new SourceMapConsumerV3();
+            section.parse(sourceMapString);
+            section.visitMappings(sourceMapGenerator::addMapping);
+
+            Iterator extensions = section.getExtensions().entrySet().iterator();
+
+            while (extensions.hasNext()) {
+              Map.Entry<String, Object> entry = (Map.Entry) extensions.next();
+              String extensionKey = (String) entry.getKey();
+              sourceMapGenerator.addExtension(extensionKey, entry.getValue());
+            }
+
+            ResourceLoader resourceLoader = ResourceLoaders.fromContextClassLoader();
+
+            Map<String, EmittedArtifact> generatedSources = new java.util.HashMap<>();
+            artifacts.find(EmittedArtifact.class)
+                .forEach(emittedArtifact -> {
+                    if (Visibility.Source == emittedArtifact.getVisibility()) {
+                      generatedSources.put(emittedArtifact.getPartialPath(), emittedArtifact);
+                    }
                 });
+
+            StandardLinkerContext stdContext = (StandardLinkerContext) context;
```

Comment:
As long as there is a cast here, might as well just access the module itself just once, rather than each time through the loop?